src/commands/LoveCalc.js

```diff
@@ -0,0 +1,105 @@
+'use strict';
+
+const { Command } = require('@dyno.gg/dyno-core');
+const superagent = require('superagent');
+
+function isText(string) { // check that the input is text-number without extra symbol + max length 30 char
+	return /^[a-zA-Z0-9]{1,30}$/.test(string);
+}
+
+function filterName(string) {
+	return /[a-zA-Z0-9]{1,30}/g.exec(string)
+}
+
+class LoveCalc extends Command {
+	constructor(...args) {
+		super(...args);
+
+		this.aliases = ['lovecalc', 'love'];
+		this.module = 'Fun';
+		this.description = 'Get your love power with someone';
+		this.usage = 'love [user] [user]\nlove [user]';
+		this.example = 'love KhaaZ Dyno\nlove Dyno';
+		this.cooldown = 5000;
+		this.expectedArgs = 1;
+		this._apiKey = this.config.mashapeAPI.key;
+	}
+
+	async execute( { message,args } ) {
+
+		let user1;
+		let user2;
+
+		// Mention, id, username, nickname => retrieve guild member
+		if (args[0]) {
+			user1 = this.resolveUser(guild, args[0]);
+		} else {
+			return this.error(message.channel, 'You need to give at least one name to test!');
+		}
+
+		if (args[1]) {
+			user2 = this.resolveUser(guild, args[1]);
+		} else {
+			user2 = message.user.username; // 1 arg = test with the user who callsed the command
+		}
+
+		// get guild Member username OR get a valid text out of the input
+		if (user1) {
+			user1 = user1.username;
+		} else if ( args[0].isText(string) ) {
+			user1 = args[0];
+		} else {
+			user1 = filterName(args[0]);
+			if (user1) {
+				user1 = user1[0];
+			}
+		}
```

Comment:
I think this could be simplified, but there's also a concern that Discord usernames don't always meet your filter criteria and you aren't validating them at all.

It would probably be better to normalize each value to a string then validate and/or filter that string here and for user2 below.